docs/root/intro/arch_overview/outlier.rst

```diff
@@ -70,38 +70,48 @@ with the minimum required request volume in an interval is less than the
 :ref:`outlier_detection.success_rate_minimum_hosts<envoy_api_field_cluster.OutlierDetection.success_rate_minimum_hosts>`
 value.
 
+.. _arch_overview_outlier_detection_logging:
+
 Ejection event logging
 ----------------------
 
 A log of outlier ejection events can optionally be produced by Envoy. This is extremely useful
 during daily operations since global stats do not provide enough information on which hosts are
-being ejected and for what reasons. The log uses a JSON format with one object per line:
+being ejected and for what reasons. The log is structured as protobuf-based dumps of `OutlierDetectionEvent messages <envoy_api_msg_cluster.OutlierDetectionEvent>`
 
 .. code-block:: json
 
   {
-    "time": "...",
+    "type": "...",
+    "timestamp": "...",
     "secs_since_last_action": "...",
-    "cluster": "...",
+    "cluster_name": "...",
```

Comment:
My main question here is what the compatibility story is. I still don't know enough myself about how Envoy is typically deployed to know whether this might break existing installations. But I'll defer to senior maintainers about whether the changing of some fields, and the changing of format from json to proto, requires some kind of transition plan for installs.

Reply:
Having very little context here (I'd love a tl;dr on what the impact will be for folks consuming the outlier events) it looks like we're changing the format of a longstanding feature, and I'd think that would fall under the breaking change "we should deprecate and give users 3 months to adapt to the new system before deleting the old" policy.

Reply:
Yeah, I think this is a gray area. We probably could get away with outright changing this, but it's not the nice thing to do since we don't know who is consuming these logs as is. 

Is there any chance to just name the fields what they were before to avoid the deprecation dance even if the names aren't completely optimal?

The flip side of this is that since we are calling the data protos alpha anyway, we aren't really guaranteeing the format won't change moving forward, so maybe we can just change it...

Reply:
Another general comment here is that I would actually remove almost everything on this docs page, and instead link into the auto generated proto docs for this structure. Basically, all the docs should live there now for field details.

Reply:
@mattklein123 would opt for consistent naming of most common fields in proto log messages across modules. In health check module we have field `cluster_name`: https://github.com/envoyproxy/envoy/blob/master/api/envoy/data/core/v2alpha/health_check_event.proto#L22
One use case is when customer is having issue/incident with cluster A and goes to aggregated logging tool to get more insights. Usually such aggregated logging tools provide possibility to filter log messages by specific field+value. So customer will have to specify compound filter like `cluster=A and cluster_name=A`. Also having different names for same concepts may introduce confusion.

Reply:
I'm OK just going ahead and changing the names here w/o deprecation, especially given that we are calling the new output alpha and subject to change anyway. I just don't think very many people are looking at this output and those that are can likely pretty easily reconfigure. @envoyproxy/maintainers thoughts/objections?

Reply:
If it's marked alpha, there is no need for a deprecation cycle.